### PR TITLE
API: Enable authentication & authorization

### PR DESCRIPTION
**Pull request by Hwesta**
Enable BasicAuthentication, ApiKeyAuthentication and DjangoAuthorization instead of the stub Authentication/Authorization previously.  Create API keys for every user and display them in the GUI.

TODO:
- [x] Update API tests to handle auth.

### CODE REVIEW

File: storage_service/templates/administration/user_form.html [hunk 1]
```diff
@@ -23,5 +23,11 @@
     </form>
     {% endif %}
 
+    {% if edit_user and edit_user.api_key%}
```
Comment (sevein) [inline storage_service/templates/administration/user_form.html]: Cosmetic: missing whitespace?

File: storage_service/locations/api/resources.py [hunk 2]
```diff
@@ -17,9 +17,8 @@
 # Third party dependencies, alphabetical
 from annoying.functions import get_object_or_None
 import bagit
-from tastypie.authentication import (BasicAuthentication, ApiKeyAuthentication,
-    MultiAuthentication, Authentication)
-from tastypie.authorization import DjangoAuthorization, Authorization
+from tastypie.authentication import BasicAuthentication, ApiKeyAuthentication,MultiAuthentication
```
Comment (sevein) [inline storage_service/locations/api/resources.py]: Cosmetic: whitespace after comma

File: storage_service/templates/administration/user_form.html (hunk 1 above)
Comment (Hwesta) [inline storage_service/templates/administration/user_form.html]: Fixed, thanks.

File: storage_service/locations/api/resources.py (hunk 2 above)
Comment (Hwesta) [inline storage_service/locations/api/resources.py]: Fixed, thanks.

**Comment by sevein** [comment]
Nice! 👍

**Comment by sevein** [comment]
👍

**Comment by Hwesta** [comment]
Also cherry-picked to 0.8 as 48c2e89240871be720e3563c94e8687623be6bd3

**Pull request merged**